### PR TITLE
fix: improve GLM presets, ideation auth, and Insights env

### PR DESCRIPTION
**Pull request by StillKnotKnown**
## Summary
- Add GLM (Global/China) presets and improve API profile dialog layout
- Align ideation auth flow with API profiles
- Expand Insights env setup (PYTHONPATH merge) and add tests

## Testing
- env -u NODE_ENV npm test -- src/main/__tests__/insights-config.test.ts
- env -u NODE_ENV npm test -- src/renderer/components/settings/ProfileEditDialog.test.tsx
- env -u NODE_ENV npm test -- src/renderer/components/settings/ModelSearchableSelect.test.tsx
- env -u NODE_ENV npm test -- src/renderer/components/ideation/hooks/__tests__/useIdeationAuth.test.ts
- env -u NODE_ENV npm test -- src/renderer/components/ideation/hooks/__tests__/useIdeation.test.ts
- npm run lint (warnings only, pre-existing)
- npm run typecheck

## Screenshots
- Not included (UI behavior verified locally)

## Summary by CodeRabbit

* **New Features**
  * API provider presets (Anthropic, OpenRouter, Groq, GLM variants) with English/French translations and localized model picker text.

* **Bug Fixes**
  * Model dropdown now renders above the input.
  * Prevent deletion of the active profile.

* **Improvements**
  * Profile editor: responsive two-column layout, preset selector, improved API key UX and focus handling.
  * Improved environment propagation so tooling and subprocesses receive the correct runtime env.

* **Tests**
  * Expanded unit tests and added a DOM polyfill and Sentry test mocks for consistent test behavior.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by AndyMik90 — CHANGES_REQUESTED** [review]
## 🤖 Auto Claude PR Review

### Merge Verdict: 🟠 NEEDS REVISION
4 issue(s) must be addressed (2 required, 2 recommended), 2 suggestions

### Risk Assessment
| Factor | Level | Notes |
|--------|-------|-------|
| Complexity | High | Based on lines changed |
| Security Impact | None | Based on security findings |
| Scope Coherence | Good | Based on structural review |

### Findings Summary
- **High**: 2 issue(s)
- **Medium**: 2 issue(s)
- **Low**: 2 issue(s)

---
_Generated by Auto Claude PR Review_

### Findings (6 selected of 6 total)

#### 🟠 [HIGH] Hardcoded strings violate i18n requirements
📁 `apps/frontend/src/renderer/components/settings/ProfileList.tsx:134`

ProfileList.tsx contains 20+ hardcoded English strings including 'API Profiles', 'No API profiles configured', 'Add Profile', 'Switch to OAuth', 'Delete Profile?', and tooltip text. Per CLAUDE.md: 'CRITICAL: Always use i18n translation keys for all user-facing text in the frontend.' Other settings components (AgentProfileSettings, GeneralSettings, DevToolsSettings) consistently use useTranslation.

**Suggested fix:**
```
Add useTranslation hook and replace all hardcoded strings with translation keys from settings.json. Add corresponding keys to en/settings.json and fr/settings.json under a 'profiles' section.
```

#### 🟠 [HIGH] Dialog contains 25+ hardcoded user-facing strings
📁 `apps/frontend/src/renderer/components/settings/ProfileEditDialog.tsx:308`

ProfileEditDialog.tsx lacks i18n support. Hardcoded strings include: 'Edit Profile', 'Add API Profile', all form labels ('Name', 'Preset', 'Base URL', 'API Key'), placeholders, button text ('Test Connection', 'Save Profile', 'Cancel'), validation errors ('Name is required', 'Invalid URL format'), and toast messages. This violates the project's explicit i18n requirement in CLAUDE.md.

**Suggested fix:**
```
Import useTranslation from 'react-i18next', add translation keys for all user-facing strings to en/settings.json and fr/settings.json, and replace hardcoded strings with t() function calls.
```

#### 🟡 [MEDIUM] Duplicated authentication check logic
📁 `apps/frontend/src/renderer/components/ideation/hooks/useIdeationAuth.ts:34`

The fallback logic for checking API profiles is duplicated between performCheck (lines 34-45) and checkAuth (lines 69-77). Both functions perform identical steps: check store activeProfileId, if empty call IPC getAPIProfiles, compute hasAPIProfile. This creates maintenance burden where bug fixes must be applied twice.

**Suggested fix:**
```
Extract the duplicated logic into a shared helper function:

const checkAPIProfile = async (profileId: string | null): Promise<boolean> => {
  let hasAPIProfile = Boolean(profileId && profileId !== '');
  if (!hasAPIProfile) {
    const profilesResult = await window.electronAPI.getAPIProfiles();
    hasAPIProfile = Boolean(profilesResult.success && profilesResult.data?.activeProfileId && profilesResult.data.activeProfileId !== '');
  }
  return hasAPIProfile;
};
```

#### 🟡 [MEDIUM] Provider presets should be in shared/constants
📁 `apps/frontend/src/renderer/components/settings/ProfileEditDialog.tsx:144`

The presets array containing provider configurations (Anthropic, OpenRouter, Groq, GLM Global, GLM China) is defined inline within the component. The codebase pattern for configuration data is to define it in apps/frontend/src/shared/constants/ (e.g., AVAILABLE_MODELS, THINKING_LEVELS, DEFAULT_AGENT_PROFILES are in models.ts).

**Suggested fix:**
```
Create apps/frontend/src/shared/constants/api-profiles.ts exporting API_PROVIDER_PRESETS, then import in ProfileEditDialog.tsx. Add export to apps/frontend/src/shared/constants/index.ts.
```

#### 🔵 [LOW] Test file has limited coverage - only one test case
📁 `apps/frontend/src/main/__tests__/insights-config.test.ts:47`

The insights-config.test.ts file contains only a single test case verifying getProcessEnv() merges environment variables correctly. Missing edge cases: error handling when getAPIProfileEnv fails, behavior when Python environment unavailable, path delimiter handling on Windows vs Unix, behavior when loadAutoBuildEnv returns empty object.

**Suggested fix:**
```
Add additional test cases for error scenarios and edge cases, particularly for getAPIProfileEnv failures and missing Python environment.
```

#### 🔵 [LOW] Test file has limited coverage - only two test cases
📁 `apps/frontend/src/renderer/components/ideation/hooks/__tests__/useIdeation.test.ts:62`

The useIdeation.test.ts file contains only two basic test cases (auth modal prompt when token missing, generate when token present). Missing: error handling during generation, loading state management, config changes triggering regeneration, listener setup/cleanup, edge cases where auth state changes during generation.

**Suggested fix:**
```
Add comprehensive test cases including: cleanup listeners on unmount test, generation error handling test, loading state transitions.
```

---
*This review was generated by Auto Claude.*

File: apps/frontend/src/shared/constants/api-profiles.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    id: 'glm-global',
+    baseUrl: 'https://api.z.ai/api/anthropic',
+    labelKey: 'settings:apiProfiles.presets.glmGlobal'
+  },
+  {
+    id: 'glm-cn',
+    baseUrl: 'https://open.bigmodel.cn/api/paas/v4',
+    labelKey: 'settings:apiProfiles.presets.glmChina'
+  }
+] as const;
```
[the remark this answers is not in the document]
Comment (StillKnotKnown) [reply apps/frontend/src/shared/constants/api-profiles.ts]: There are two variant of the China and Global endpoints, OpenAI compatable and Anthropic compatable. endpoints implemented in code is aligned to claude code cli usage using the Anthropic endpoints

File: apps/frontend/src/renderer/components/settings/ModelSearchableSelect.tsx [hunk 2]
```diff
@@ -254,15 +259,18 @@ export function ModelSearchableSelect({
 
       {/* Dropdown panel - only show when we have models to display */}
       {isOpen && !isLoading && !modelDiscoveryNotSupported && models.length > 0 && (
-        <div className="absolute z-50 w-full mt-1 bg-background border rounded-md shadow-lg max-h-60 overflow-hidden flex flex-col">
+        <div
+          className="absolute z-50 w-full bottom-full mb-1 bg-background border rounded-md shadow-lg max-h-60 overflow-hidden flex flex-col"
+          data-testid="model-select-dropdown"
+        >
```
[the remark this answers is not in the document]
Comment (StillKnotKnown) [reply apps/frontend/src/renderer/components/settings/ModelSearchableSelect.tsx:265]: it does not pass the boundaries as it limits to certain height with search components and scroll bars that are consistent between all dropdowns

**Comment by StillKnotKnown** [comment]
Lint fix pushed: repaired malformed env return in insights config. Local eslint passes with "npm run lint -- --quiet".

**Comment by StillKnotKnown** [comment]
Tests run locally:\n- npm run lint (0 errors, 346 warnings)\n- npm run typecheck\n- npm test\n\nAdded Sentry module mocks for Vitest + type stubs to unblock renderer tests.

**Comment by StillKnotKnown** [comment]
Sentry mock/type refactor pushed: shared mock module + shared type definitions in d.ts.\n\nTests run locally:\n- npm run lint (0 errors, 346 warnings)\n- npm run typecheck\n- npm test

**Review by AndyMik90 — APPROVED** [review]
